### PR TITLE
TFM: Add Kconfig to set MCUBOOT_IMAGE_NUMBER

### PR DESCRIPTION
**Pull request by oyvindronningstad**
~Based on #419~

### CODE REVIEW

**Comment by mbolivar-nordic** [comment]
Please rebase this PR if still relevant; master has been rebased.

**Review by hakonfam — APPROVED** [review]

**Comment by mbolivar-nordic** [comment]
@hakonfam you've still got -1 on https://github.com/nrfconnect/sdk-nrf/pull/3790/files, so I'm assuming you don't want this merged yet.

**Comment by hakonfam** [comment]
> @hakonfam you've still got -1 on [nrfconnect/sdk-nrf#3790](https://github.com/nrfconnect/sdk-nrf/pull/3790), so I'm assuming you don't want this merged yet.

I guess we should wait with merging until the parent PR is approved at least.

**Review by ioannisg — APPROVED** [review]
Looks fine to me.
@oyvindronningstad can you take a look at the related upstream PRs, where I am sure I have left some comments in the PR reviews? MAybe you can bring some of those here.

**Review by tejlmand — CHANGES_REQUESTED** [review]
I propose a choice, to make it clear what options the user has, and to make the default follow how Zephyr is configured.

File: modules/Kconfig.tfm [hunk 1]
```diff
@@ -78,6 +78,12 @@ config TFM_PROFILE
 	  profile_medium and profile_small. The default profile does not need
 	  to have this configuration set.
 
+config TFM_CMAKE_BUILD_TYPE
+	string "The build type sent to the TFM build system"
+	default "RelWithDebInfo"
+	help
+	  Choose from Debug, Release, RelWithDebInfo, and MinSizeRel
```
[the remark this answers is not in the document]
Comment (oyvindronningstad) [reply modules/Kconfig.tfm]: Done

File: modules/Kconfig.tfm [hunk 2]
```diff
@@ -78,6 +78,28 @@ config TFM_PROFILE
 	  profile_medium and profile_small. The default profile does not need
 	  to have this configuration set.
 
+choice
+	bool
+	prompt "The build type sent to the TFM build system"
```
Comment (tejlmand) [inline modules/Kconfig.tfm:83]: Just did a UI try of this:
image

Maybe shorten this to:
image

```suggestion
	prompt "TFM build type"
```

**Review by tejlmand — APPROVED** [review]
Looks good.
Consider shorten the build type prompt.
